src/Modules/UngradedModules.jl

```diff
@@ -0,0 +1,3653 @@
+    r.gens = gens
+    if singular_generators(gens).isGB
+      r.std_basis = r.gens
+    end
+    return r
+  end
+
+  function SubModuleOfFreeModule{L}(F::FreeMod{L}, A::MatElem{L}) where {L}
+    r = new{L}()
+    r.F = F
+    O = [FreeModElem(sparse_row(A[i,:]), F) for i in 1:nrows(A)]
+    r.gens = ModuleGens(O, F)
+    r.matrix = A
+    return r
+  end
+end
+
+@doc Markdown.doc"""
+    SubModuleOfFreeModule(F::FreeMod{R}, gens::Vector{<:FreeModElem}) where {R}
+
+Construct the submodule of `F` generated by the elements of `gens` (the elements of 
+`gens` must live in `F`).
+"""
+SubModuleOfFreeModule(F::FreeMod{R}, gens::Vector{<:FreeModElem}) where {R} = SubModuleOfFreeModule{R}(F, gens)
+
+SubModuleOfFreeModule(F::FreeMod{R}, singular_module::Singular.smodule) where {R} = SubModuleOfFreeModule{R}(F, singular_module)
+
+@doc Markdown.doc"""
+    SubModuleOfFreeModule(F::FreeMod{R}, gens::ModuleGens{R}) where {R}
+
+Construct the submodule of `F` generated by `gens`.
+"""
+SubModuleOfFreeModule(F::FreeMod{R}, gens::ModuleGens{R}) where {R} = SubModuleOfFreeModule{R}(F, gens)
+
+@doc Markdown.doc"""
+    SubModuleOfFreeModule(F::FreeMod{L}, A::MatElem{L}) where {L}
+
+Construct the submodule generated by the rows of `A`. The embedding free
+module is `F`. In particular, `rank(F) == ncols(A)` must hold.
+"""
+SubModuleOfFreeModule(F::FreeMod{L}, A::MatElem{L}) where {L} = SubModuleOfFreeModule{L}(F, A)
+
+@doc Markdown.doc"""
+    SubModuleOfFreeModule(A::MatElem{L}) where {L} 
+
+Construct the submodule generated by the rows of `A`.
+Note: The embedding free module of the submodule is constructed by the function and therefore
+not compatible with free modules that are defined by the user or other functions. 
+If you need compatiblity use `SubModuleOfFreeModule(F::FreeMod{L}, A::MatElem{L}) where {L}`.
+"""
+function SubModuleOfFreeModule(A::MatElem{L}) where {L} 
+  R = base_ring(A)
+  F = FreeMod(R, ncols(A))
+  return SubModuleOfFreeModule{L}(F, A)
+end
+
+function Base.getindex(M::SubModuleOfFreeModule, i::Int)
+  return oscar_generators(M.gens)[i]
+end
+
+@doc Markdown.doc"""
+    iszero(M::SubModuleOfFreeModule)
+
+Check if `M` is zero.
+"""
+function iszero(M::SubModuleOfFreeModule)
+  return iszero(M.gens)
+end
+
+@doc Markdown.doc"""
+    base_ring(M::SubModuleOfFreeModule)
+
+Return the base ring of `M`.
+"""
+function base_ring(M::SubModuleOfFreeModule)
+  return base_ring(M.F)
+end
+
+@doc Markdown.doc"""
+    std_basis(submod::SubModuleOfFreeModule)
+
+Compute a standard basis of `submod`.
+"""
+function std_basis(submod::SubModuleOfFreeModule)
+  if !isdefined(submod, :std_basis)
+    submod.std_basis = groebner_basis(submod.gens)
+  end
+  return submod.std_basis
+end
+
+@doc Markdown.doc"""
+    matrix(submod::SubModuleOfFreeModule)
+
+Return the generators of `submod` in matrix-form, that is the rows of the 
+matrix generate `submod`.
+"""
+function matrix(submod::SubModuleOfFreeModule)
+  if !isdefined(submod, :matrix)
```

Comment:
I find this rather problematic. We want to move away from representing modules as matrices. Perhaps internally, deep down, this is useful, but I am very reluctant of exposing this to the user in such a prominent (and even documented) fashion. Perhaps if the function was called `_submod_as_matrix_internal`...

Why do you need this?